lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Context/RootContextLauncher.cs

```diff
@@ -39,17 +40,24 @@ internal sealed class RootContextLauncher
         private ISet<object> _services;
         private ContextRuntime _rootContext;
 
-        public RootContextLauncher(string id, IConfiguration contextConfiguration,
-            Optional<IConfiguration> rootServiceConfig, Optional<IConfiguration> rootTaskConfig, IHeartBeatManager heartbeatManager)
+        public RootContextLauncher(
+            IConfiguration contextConfiguration, 
+            IConfiguration rootServiceConfig, 
+            Optional<IConfiguration> rootTaskConfig, 
+            IHeartBeatManager heartbeatManager)
         {
-            Id = id;
             _rootContextConfiguration = contextConfiguration;
-            _rootServiceInjector = InjectServices(rootServiceConfig);
+            _rootServiceInjector = TangFactory.GetTang().NewInjector(rootServiceConfig);
+            Id = _rootServiceInjector
+                .ForkInjector(contextConfiguration)
+                .GetNamedInstance<ContextConfigurationOptions.ContextIdentifier, string>();
+            _services = _rootServiceInjector.GetNamedInstance<ServicesSet, ISet<object>>();
+            Logger.Log(Level.Verbose, string.Format(CultureInfo.InvariantCulture, "injected service(s)"));
             _rootServiceInjector.BindVolatileInstance(GenericType<IHeartBeatManager>.Class, heartbeatManager);
```

Comment:
Because the Evaluator is still reading in `DriverConfiguration`, and not using `BindVolatileInstance` causes conflicts. This will be removed with 217.